routing/message/handlers/passthrough/device_handler.go

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package passthrough
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/eclipse-kanto/aws-connector/config"
+	"github.com/eclipse-kanto/aws-connector/routing/message/handlers"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/eclipse-kanto/suite-connector/connector"
+	"github.com/eclipse-kanto/suite-connector/routing"
+	"github.com/eclipse/ditto-clients-golang/protocol"
+)
+
+const (
+	deviceHandlerName = "passthrough_device_handler"
+	topicsLocal       = "event/#,e/#,telemetry/#,t/#"
+
+	valueAttributesTag = "attributes"
+	valueFeaturesTag   = "features"
+	valuePropertiesTag = "properties"
+	valueDefinitionTag = "definition"
+	valueStateTag      = "state"
+	valueReportedTag   = "reported"
+
+	// Used to update attributes of root thing.
+	topicRootShadow = "$aws/things/%s/shadow/%s"
+	// Used to update feature of root thing or attributes of child thing.
+	topicNamedShadow = "$aws/things/%s/shadow/name/%s/%s"
+	// Used to update feature of child thing.
+	topicComplexNamedShadow = "$aws/things/%s/shadow/name/%s:%s/%s"
+
+	topicUpdate = "update"
```

Comment:
Update performs a merge in the shadow document. When a container is deleted, the container management SoftwareUpdatable feature send Ditto installedDependencies property modify with the deleted container missing. As in the AWS shadows a merge is performed the deleted container persist in the installedDependency part of the shadow document. Eventually this leads to over the limit shadow document:
```
{
    "timestamp": "2023-09-25 13:15:20.495",
    "logLevel": "ERROR",
    "traceId": "8549dcc6-87b6-a5db-75ab-f442872a02fa",
    "accountId": "797412740850",
    "status": "Failure",
    "eventType": "UpdateThingShadow",
    "protocol": "MQTT",
    "deviceShadowName": "edge:SF3-C-0006W",
    "topicName": "$aws/things/edge:SF3-C-0006W/shadow/name/edge:containers:SoftwareUpdatable/update",
    "details": "Json document too large.  Size is 8359, max allowed: 8192"
}
```
Unfortunately I could not see an easy solution to this problem.

Reply:
We discussed some potential solutions of the problem. @yonko-gospodinov will investigate, whether it is possible to synchronization the 'delete' + 'merge' messages on the edge side via 'delete' response with the current suite-connector API. I will check whether this problem impacts only SoftwareUpdatable features and how much is the effort to switch to Ditto merge messages for the impacted features.

The further progress will be traced in a separate issue - #3.